NuRadioMC/SignalProp/CPPAnalyticRayTracing/analytic_raytracing.cpp

```diff
@@ -285,20 +285,34 @@ double dt_freq (double t, void *p){
 }
 
 double get_attenuation_along_path(double pos[2], double pos2[2], double C0,
-		double frequency, double n_ice, double delta_n, double z_0, int model){
+		double frequency, double n_ice, double delta_n, double z_0, int model) {
+
 	double x2_mirrored[2]={0.};
 	get_z_mirrored(pos,pos2,C0,x2_mirrored, n_ice, delta_n, z_0);
 
 	gsl_integration_workspace *w = gsl_integration_workspace_alloc(2000);
 	gsl_function F;
 	F.function = &dt_freq;
-	struct dt_freq_params params = {C0,frequency, n_ice, delta_n, z_0, model};
+	struct dt_freq_params params = {C0, frequency, n_ice, delta_n, z_0, model};
 	F.params=&params;
 
 	double result, error;
-	double epsrel = 1.e-7; //small initial absolute error
-	int max_badfunc_tries=6;
-	int num_badfunc_tries=0;
+	double epsabs = 0; // 0 -> only take into account relative error
+	double epsrel = 1.e-7; // small initial relative error
+	double max_epsrel = 1.e-5; // max excepted relative error
+
+	// We have to tolerate larger uncertainties for the GL3 model because it's rougthness is
+	// a challange for the integration
+	if (model == 5) {
+		epsabs = 1.e-4;
+		epsrel = 1.e-5;
+		max_epsrel = 1.e-3;
+	}
+	
+	int max_badfunc_tries = 5;
+	int num_badfunc_tries = 0;
+	double delta_epsrel = (max_epsrel - epsrel) / max_badfunc_tries; // small initial relative error
+
```

Comment:
I think there's something to say for the previous logarithmic increase in the maximum error, rather than the linear one here - you're going from 1e-7 to 2e-6 immediately, now, which is quite a large jump. But this is probably a bit nitpicky.

Reply:
done